pumps/mongo.go

```diff
@@ -103,6 +107,115 @@ func (m *MongoPump) Init(config interface{}) error {
 	return nil
 }
 
+func (m *MongoPump) capCollection() {
+
+	var colName = m.dbConf.CollectionName
+	var colCapMaxSizeBytes = m.dbConf.CollectionCapMaxSizeBytes
+
+	exists, err := m.collectionExists(colName)
+	if err != nil {
+		// TODO Handle Error
+
+		return
+	}
+
+	if !exists {
+		if err := m.createCappedCollection(colName, colCapMaxSizeBytes); err != nil {
+			// TODO handle error
+
+			return
+		}
+
+		return
+	}
+
+	// If colCapMaxSizeBytes == 0, uncap the collection and return
+	if colCapMaxSizeBytes == 0 {
+		m.uncapCollection(colName)
+
+		return
+	}
+
+	m.resizeCappedCollection(colName, colCapMaxSizeBytes)
+}
+
+// collectionExists checks to see if a collection name exists in the db.
+func (m *MongoPump) collectionExists(name string) (bool, error) {
+	sess := m.dbSession.Copy()
+	defer sess.Close()
+
+	colNames, err := sess.DB("").CollectionNames()
+	if err != nil {
+		log.WithFields(logrus.Fields{
+			"prefix": mongoPrefix,
+		}).Error("Unable to get column names: ", err)
+
+		return false, err
```

Comment:
nit, but do `return false, fmt.Errorf("unable to get column names: %v", err)` and log in the caller

(it's generally frowned upon to both log and return an error, even if the old code does it quite often)